test/Microsoft.AspNetCore.Mvc.Core.Test/Infrastructure/ObjectResultExecutorTest.cs

```diff
@@ -75,177 +43,110 @@ public void SelectFormatter_WithNoProvidedContentType_DoesConneg()
-                    // The accept header does not match anything in ObjectResult.ContentTypes.
-                    // The first formatter that can write the result gets to choose the content type.
-                    { contentTypes, "text/custom", "application/json" },
-
-                    // Accept header matches ObjectResult.ContentTypes, but no formatter supports the accept header.
-                    // The first formatter that can write the result gets to choose the content type.
-                    { contentTypes, "text/xml", "application/json" },
-
-                    // Filters out Accept headers with 0 quality and selects the one with highest quality.
+                    // Empty Accept header value, should select based on ObjectResult.ContentTypes.
+                    new ConnegScenario
+                    {
+                        ContentTypes = { "text/plain", "text/xml", "application/json"},
+                        AcceptHeaderValue = "",
+                        ResultingContentType = new StringSegment("application/json"),
+                        Formatters = { cannotWriteFormatter, jsonFormatter },
+                        ResultingFormatter = jsonFormatter,
+                    },
+                    // Null Accept header value, should select based on ObjectResult.ContentTypes.
+                    new ConnegScenario 
+                    { 
+                        ContentTypes = { "text/plain", "text/xml", "application/json"},
+                        AcceptHeaderValue = null, 
+                        ResultingContentType = new StringSegment("application/json"),
+                        Formatters = { cannotWriteFormatter, jsonFormatter },
+                        ResultingFormatter = jsonFormatter,
+                    },
+                    // Respects Accept header value by selecting the most preferable formatter
+                    new ConnegScenario
+                    {
+                        ContentTypes = { "text/plain", "text/xml", "application/json"},
+                        AcceptHeaderValue = "text/plain;q=0.3, text/xml;q=0.5, text/custom;q=0.0, application/json;q=0.4",
+                        ResultingContentType = new StringSegment("text/xml"),
+                        Formatters = { cannotWriteFormatter, jsonFormatter, xmlFormatter },
+                        ResultingFormatter = xmlFormatter,
+                    },
+                    // Respects Accept header value by failing to negotiate an unsupported media type
+                    new ConnegScenario
+                    {
+                        ContentTypes = { "text/plain", "text/xml", "application/json"},
+                        AcceptHeaderValue = "application/vnd.custom",
+                        ResultingContentType = default(StringSegment),
+                        Formatters = { cannotWriteFormatter, jsonFormatter },
+                        ResultingFormatter = null,
+                        ReturnHttpNotAcceptable = true,
```

Comment:
The status code is set by `ExecuteAsync` rather than `SelectFormatter`, and it is already verified by the `ExecuteAsync_NoFormatterFound_Returns406` test.

I will add explicit values for `ReturnHttpNotAcceptable` to verify that the settings does not negatively impact cases where there is an acceptable formatter.